web_sdk/sdk_rewriter.dart

```diff
@@ -130,13 +137,25 @@ void main(List<String> arguments) {
     preprocessor = (String source) => preprocessPartFile(source, libraryName!);
     replacementPatterns = generatePartsPatterns(libraryName, isPublic);
   }
-  for (final String inputFilePath in results['source-file'] as Iterable<String>) {
-    String pathSuffix = inputFilePath.substring(inputDirectoryPath.length);
+
+  final List<String> inputFiles = <String>[];
+  final List<FileSystemEntity> entries = Directory(inputDirectoryPath).listSync(
```

Comment:
I'm assuming enumerating files in Dart is faster than in gn?

Reply:
It may or may not be (though my guess is that this *is* faster in Dart than Python). The advantage of moving the directory listing here is that instead of happening unconditionally on every GN invocation, it will only happen when the target needs to be rebuilt during the build, and when it is rebuilt it will be in parallel with the rest of the build, rather than as the long-pole of the GN step.